### PR TITLE
scoring: add test groups

### PR DESCRIPTION
**Pull request by sirupsen**
Previously a `Task` had `TestCases`, and we'd do scoring by summing the number of completed test case, and the total score would be the % of total tests completed.

Instead, this implements Test Groups. Each test group has a set of tasks, and you only get the points for that group if you complete all the tasks within it.

@xldenis

### CODE REVIEW

**Comment by haileys** [comment]
shout out to turbomerging without review

**Comment by sirupsen** [comment]
For review @charliesome

**Comment by sirupsen** [comment]
aren't you supposed to be on a plane

File: app/models/submission.rb [hunk 1]
```diff
@@ -30,14 +30,21 @@ def set_competition
     self.competition = task.competition if task.competition.open?
   end
 
-  # Returns the amount of points for an IOI-style task submission.
-  def points
-    raise "Not an IOI style task." unless task.scoring == "ioi"
-    total  = body.count.to_f
-    raise "Task has no tests." unless total > 0
+  # FIXME: LOL
```
Comment (haileys) [inline app/models/submission.rb:33]: lol!

**Comment by haileys** [comment]
LGTM :+1:

**Pull request merged**